src/lib/types.ts

```diff
@@ -951,6 +1017,21 @@ export type AuthMFAChallengeResponse =
       }
       error: null
     }
+  | {
+      data: {
+        /** ID of the newly created challenge. */
+        id: string
+
+        /** Factor Type which generated the challenge */
+        type: 'webauthn'
+
+        /** Timestamp in UNIX seconds when this challenge will no longer be usable. */
+        expires_at: number
+        credential_creation_options: { publicKey: PublicKeyCredentialCreationOptionsJSON }
+        credential_request_options: { publicKey: PublicKeyCredentialRequestOptionsJSON }
+      }
+      error: null
+    }
   | { data: null; error: AuthError }
 
 export type AuthMFAListFactorsResponse =
```

Comment:
Decide whether to include aaguid to image / authenticator type mapping for user: https://github.com/passkeydeveloper/passkey-authenticator-aaguids

or to include it in documentation

Reply:
Docs is better, we can add that in a follow up based on feedback.